src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,664 @@
+    }
+
+    function testRemoveStrategiesConsideredAndMultipliers_Valid(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers,
+        uint256[] memory indicesToRemove
+    ) public {
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+        // take indices modulo length
+        for (uint256 i = 0; i < indicesToRemove.length; i++) {
+            indicesToRemove[i] = indicesToRemove[i] % strategiesAndWeightingMultipliers.length;
+        }
+        indicesToRemove = _removeDuplicatesUint256(indicesToRemove);
+        cheats.assume(indicesToRemove.length > 0);
+        cheats.assume(indicesToRemove.length < strategiesAndWeightingMultipliers.length);
+        indicesToRemove = _sortInDescendingOrder(indicesToRemove);
+
+        // create the quorum
+        uint8 quorumNumber = voteWeigher.quorumCount();
+        cheats.startPrank(serviceManagerOwner);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+
+        // remove certain strategies
+        // make sure events are emmitted
+        for (uint i = 0; i < indicesToRemove.length; i++) {
+            cheats.expectEmit(true, true, true, true, address(voteWeigher));
+            emit StrategyRemovedFromQuorum(quorumNumber, strategiesAndWeightingMultipliers[indicesToRemove[i]].strategy);
+        }
+        voteWeigher.removeStrategiesConsideredAndMultipliers(quorumNumber, indicesToRemove);
+
+        // check that the strategies were removed
+        
+        // check that the strategies that were not removed are still there
+        // get all strategies and multipliers form the contracts
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliersFromContract = new IVoteWeigher.StrategyAndWeightingMultiplier[](voteWeigher.strategiesConsideredAndMultipliersLength(quorumNumber));
+        for (uint256 i = 0; i < strategiesAndWeightingMultipliersFromContract.length; i++) {
+            strategiesAndWeightingMultipliersFromContract[i] = voteWeigher.strategyAndWeightingMultiplierForQuorumByIndex(quorumNumber, i);
+        }
+
+        // remove indicesToRemove from local strategiesAndWeightingMultipliers
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliersLocal = new IVoteWeigher.StrategyAndWeightingMultiplier[](strategiesAndWeightingMultipliers.length - indicesToRemove.length);
+        uint256 j = 0;
+        for (uint256 i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            for (uint256 k = 0; k < indicesToRemove.length; k++) {
+                if (i == indicesToRemove[k]) {
+                    break;
+                }
+                if (k == indicesToRemove.length - 1) {
+                    strategiesAndWeightingMultipliersLocal[j] = strategiesAndWeightingMultipliers[i];
+                    j++;
+                }
+            }
+        }
```

Comment:
this nested loop seems particularly inefficient, and I suspect it's avoidable
tests are hanging for me and I believe this test is the (or at least _a_ culprit)

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/53/commits/2c3df9ab884f24281df9c3ef514b72b7edb086b3